### PR TITLE
[v21.0.x] backport: 21.0.1 backports and release notes

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Backport PRs for v21.0.1

## What was done?
See commits

## How Has This Been Tested?
See CI

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -35,366 +34,30 @@ reindex or re-sync the whole chain.
...
-will also initiate the launch of the Dash Platform Genesis Chain.
+Transaction Retrieval Bug
+-------------------------
 
-As the MN_RR hard fork is an [Enhanced Hard Fork](https://github.com/dashpay/dips/blob/master/dip-0023.md),
-activation is dependent on both masternodes and miners. However, as this hard fork was first implemented in v20, only
-masternodes are required to upgrade for the hard fork to activate. Once 85% of masternodes have upgraded to v21,
-an EHF message will be signed by the `LLMQ_400_85` quorum, and mined into a block. This signature serves as proof that 85% of
-active masternodes have upgraded.
+A bug that prevented retrieval of information for unmined transactions with clients that do not participate in address
```
Comment (knst) [inline doc/release-notes.md:40]: I like the very first version, but maybe change "This resulted in SPV nodes not proactively receiving TX messages" to something like  "This resulted Dash Core v21 is not sending anything but blocks to light clients" ?

I even can't find in Google what is "address relaying" in bitcoin/dash.

-----
First version:

 SPV Client Bugfix
---------------------------------------------------------
Networking backports resulted in a regression where SPV / light nodes are incorrectly treated as block_only connections.
This resulted in SPV nodes not proactively receiving TX messages. Kuddos to @HashEngineering and @knst for finding and
investigating this issue.

File: doc/release-notes.md [hunk 2]
```diff
@@ -35,366 +34,30 @@ reindex or re-sync the whole chain.
...
+relaying due to partial misclassification as a block-only connection, has been resolved. This was identified by Dash
+mobile wallets not seeing transactions until they were mined. Kudos to @HashEngineering and @knst for finding and investigating this issue.
```
Comment (knst) [inline doc/release-notes.md:42]:
```suggestion
SPV Client Bugfix
---------------------------------------------------------
Networking backports resulted in a regression where SPV / light nodes are incorrectly treated as block_only connections.
This resulted Dash Core v21 is not sending anything but blocks to light clients.
Kuddos to @HashEngineering and @knst for finding and
investigating this issue.
```

Comment (kwvg) [reply doc/release-notes.md:42]: The clients weren't fully treated as block-only (only one check was affected) nor did the bug exclusively affect light clients (it affected any client that didn't indicate it wanted to relay addresses).

[the remark this answers is not in the document]
Comment (kwvg) [reply doc/release-notes.md:42]: LGTM

**Review by kwvg — APPROVED** [review]
utACK 5619c8f5fdb06ec3762697b74360d0ace08660d5

**Review by UdjinM6 — APPROVED** [review]
utACK 5619c8f5fdb06ec3762697b74360d0ace08660d5

**Pull request merged**